Source/JavaScriptCore/runtime/DateConstructor.cpp

```diff
@@ -71,6 +71,37 @@ static double millisecondsFromComponents(JSGlobalObject* globalObject, const Arg
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    auto toIntegerOrInfinity = [](double d) {
+        return trunc(std::isnan(d) ? 0.0 : d + 0.0);
+    };
+
+    // https://tc39.es/ecma262/#sec-maketime
+    auto makeTime = [](double hour, double min, double sec, double ms) {
+        return ((hour * msPerHour + min * msPerMinute) + sec * msPerSecond) + ms;
+    };
+
+    // https://tc39.es/ecma262/#sec-makeday
+    auto makeDay = [](double year, double month, double date) {
+        double additionalYears = std::floor(month / 12);
+        double ym = year + additionalYears;
+        if (!std::isfinite(ym))
+            return PNaN;
+        double mm = month - additionalYears * 12;
+        int32_t yearInt32 = toInt32(ym);
+        int32_t monthInt32 = toInt32(mm);
+        if (yearInt32 != ym || monthInt32 != mm)
+            return PNaN;
+        double days = dateToDaysFrom1970(yearInt32, monthInt32, 1);
+        return days + date - 1;
+    };
+
+    // https://tc39.es/ecma262/#sec-makedate
+    auto makeDate = [](double day, double time) {
+        if (!std::isfinite(day) || !std::isfinite(time))
+            return PNaN;
+        return day * msPerDay + time;
+    };
```

Comment:
Aren't these operations used elsewhere in the Date API? How come we need to define them here?

Reply:
The other places are using it too, but I'm currently only focusing on Date constructor and Date.UTC, and I also currently have no idea whether the other place should do it too, or not.